ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -48,16 +49,21 @@ export default class DashboardQuickActionsCard extends Component {
           title: 'Secret path',
           subText: 'Path of the secret you want to read, including the mount. E.g., secret/data/foo.',
           buttonText: 'Read secrets',
-          // check kv version to figure out which model to use
-          model: this.selectedEngine.version === 2 ? 'secret-v2' : 'secret',
-          route: 'vault.cluster.secrets.backend.show',
+          model: 'kv/metadata',
+          route: 'vault.cluster.secrets.backend.kv.secret.details',
+          nameKey: 'path',
+          queryObject: { pathToSecret: '', backend: this.selectedEngine.id },
+          objectKeys: ['path', 'id'],
         };
       case 'Generate credentials for database':
         return {
           title: 'Role to use',
           buttonText: 'Generate credentials',
           model: 'database/role',
           route: 'vault.cluster.secrets.backend.credentials',
+          nameKey: 'id',
```

Comment:
is the id also the name? the default of name should be fine and we shouldn't need to pass `nameKey` here